libproc/lustre.c

```diff
@@ -131,6 +131,42 @@ typedef enum {
  * On EOF, functions return -1 with errno clear.
  */
 
+/*
+ * Fill-in the supplied version components with the Lustre version
+ * found in /sys.
+ *
+ * Returns -1 or less on error, 0 or greater on success.
+ */
+int
+sys_fs_lustre_version (pctx_t ctx, int *major, int *minor, int *patch,
+                       int *fix)
+{
+    int ret = -1;
+    char version[256];
+
+    if ((ret = proc_openf (ctx, PROC_FS_LUSTRE_VERSION)) < 0)
+        goto done;
+
+    if ((ret = proc_gets (ctx, NULL, version, sizeof (version))) < 0)
+        goto close;
+
+    /* first, get the numbers which must be present in a version string */
+    if ((ret = sscanf (version, "%d.%d.%d", major, minor, patch)) != 3) {
+        errno = EIO;
+        ret = -3 + ret;
```

Comment:
?

Reply:
This was borrowed from `proc_fs_lustre_version` for consistency. It's to cover the case where `sscanf` reads fewer than 3 integers (`sscanf` returns the number of items it was able to read). I can simply return `-1` in these cases if that is preferred.